lambdas/process-cri-callback/src/main/java/uk/gov/di/ipv/core/processcricallback/ProcessCriCallbackHandler.java

```diff
@@ -227,14 +227,15 @@ public APIGatewayProxyResponseEvent handleRequest(
         }
     }
 
+    @ExcludeFromGeneratedCoverageReport
     private CriCallbackRequest parseCallbackRequest(APIGatewayProxyRequestEvent input)
-            throws ParseCriCallbackRequestException {
+            throws ParseCriCallbackRequestException, HttpResponseExceptionWithErrorBody {
         try {
             var callbackRequest = objectMapper.readValue(input.getBody(), CriCallbackRequest.class);
-            callbackRequest.setIpvSessionId(input.getHeaders().get("ipv-session-id"));
+            callbackRequest.setIpvSessionId(RequestHelper.getIpvSessionId(input));
             callbackRequest.setFeatureSet(RequestHelper.getFeatureSet(input.getHeaders()));
-            callbackRequest.setIpAddress(input.getHeaders().get("ip-address"));
-
+            callbackRequest.setIpAddress(RequestHelper.getIpAddress(input));
```

Comment:
Looking at the implementation of RequestHelper.getIpvSessionId and RequestHelper. getIpAddress I think this changes the behaviour - will throw error if the header is missing. I don't think we want that, especially for ipvSessionId which may be null in the case of the cross-browser issue